.github/workflows/rust.yml

```diff
@@ -184,6 +184,8 @@ jobs:
           rust-version: stable
       - name: Run tests (excluding doctests)
         run: cargo test --profile ci --exclude datafusion-examples --exclude ffi_example_table_provider --exclude datafusion-benchmarks --workspace --lib --tests --bins --features avro,json,backtrace,integration-tests
+      - name: Run serde feature test
```

Comment:
I don't think we need to run the tests -- we just need to check that the code compiles

Perhaps we can add the check here:
https://github.com/apache/datafusion/blob/04d823b852a6b3b8dd2e5b11491f0012e41b8d7e/.github/workflows/rust.yml#L116-L118

And adjust the comments to say it works with a subset

```toml
  # cargo check datafusion to ensure that the datafusion crate can be built with only a
  # subset of the function packages enabled.
```

```toml
  # cargo check datafusion to ensure that the datafusion crate can be built with only a
  # subset of the packages enabled.
```

I'll make this change and push to your branch to save some more back and forth cycles